### ISSUE TITLE
Improve Flow Run Summary report in mHero

### ISSUE THREAD

**Issue by cloehr651** [intrahealth/emNutt#15]
The Flow Run Summary report should fulfill the request for a report that lists which contacts received messages although it doesn't seem to show the contacts for which the sending failed (just interrupted). Is there any way to add "failed" to this report?

The order of the columns makes this report a little bit harder to use for the purpose of seeing sent/failed (it seems more suited for looking at responses of individual contacts). Would it be possible for users to customize the layout (i.e. moving or even hiding columns)?

I'm also noticing a few issues with this report:
- Sorting column by date (started on, modified on, ended on) causes an error "Unable to load messages"
- It would be helpful to add a filter for workflow and maybe by name too

**Comment by njugunask** [comment]
The new feature implemented by Ally to check on the progress of the workflow during messaging is working. The report is displayed on the screen giving a percentage of the progress while updating the number of successful recipients. However, the report though has a tab of 'failed' messages is not populated. Please consider the following recommendations.

1. Have a report on the failed messages, numbers.
2. Some of the numbers might be wrong or missing altogether, can we have a list of the recipients without numbers or incomplete numbers so as to ascertain successful messages sent
3. We have approximately 21/22k contacts with phone numbers in iHRIS but on mHero, we have 58k contacts.This means that the balance has been coming from Rapidpro

Other points or issue of consideration in reporting:

1. County: No. of messages being sent out to each county
2. Cadre: Numbers reached per SMS in each cadre
3. Date: A list of all messages sent and which day the message was sent
4. Adding a Person there's an error on saving a record-not able to save
5. If contacts have been synced from Rapid pro to mHero, do we have a option to bulk remove them from the platform e.g. remove an entire group. This is important, given that groups deleted on Rapidpro are still existing on mHero, test groups on Rapidpro etc
6. Duplication of records: If a contact exisits in Rapidpro and on iHRIS, two records are created for the same person especially if the phone number format is different. This means that we will have numerous duplicates on mHero. e.g. janet wanjiku karimi,Michael Oluoch Ogalo
7.

**Comment by cloehr651** [comment]
This issue is actually many different issues so I will add details to other tickets that already exist for these issues.

**Comment by cloehr651** [comment]
@ashaban have you addressed points 1 & 2 already?

Points 3 and the entire second set are different issues that I will merge with other existing issues or create new ones.

**Comment by cloehr651** [comment]
The first three points and the last four points in the second set are not related to the "Flow Run Summary" report.

@ashaban is it possible to add other options for filtering (i.e. county and cadre) the "Flow Run Summary" report?

**Comment by njugunask** [comment]
Need to add a report for '**failed**' contacts

**Comment by cloehr651** [comment]
@njugunask is this addressed in the option to filter by send status in the flow run and message sending breakdown reports?

**Comment by njugunask** [comment]
@cloehr651, Yes. The 'failed' message report is available in the "Flow run breakdown" report. This issue can be closed. However, @IKwirica needs to ascertain and give a go-ahead to close the issue.
